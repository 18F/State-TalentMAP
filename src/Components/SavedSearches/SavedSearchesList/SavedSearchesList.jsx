import React from 'react';
import PropTypes from 'prop-types';
import SavedSearchesListResultsCard from '../SavedSearchesListResultsCard';
// import Alert from '../../Alert';
import { SAVED_SEARCH_PARENT_OBJECT, MAPPED_PARAM_ARRAY } from '../../../Constants/PropTypes';

const SavedSearchesList = ({ savedSearches, goToSavedSearch, deleteSearch, cloneSavedSearch,
mappedParams }) => {
  const savedSearchArray = [];
  savedSearches.results.slice().forEach(savedSearch => (
    savedSearchArray.push(
      <SavedSearchesListResultsCard
        savedSearch={savedSearch}
        goToSavedSearch={goToSavedSearch}
        deleteSearch={deleteSearch}
        key={savedSearch.id}
        cloneSavedSearch={cloneSavedSearch}
        mappedParams={mappedParams}
        /* pass a parentClassName that we can use from the BorderedList component */
        parentClassName="parent-list-container list-transparent"
      />,
    )
  ));
  return (
    <div className="usa-grid-full">
      {
        <div className="usa-grid-full saved-searches-list">
          {savedSearchArray.map(s => s)}
        </div>
      }
    </div>
  );
};

SavedSearchesList.propTypes = {
  savedSearches: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
  cloneSavedSearch: PropTypes.func.isRequired,
  mappedParams: MAPPED_PARAM_ARRAY.isRequired,
};

SavedSearchesList.defaultProps = {
  mappedParams: [],
};

export default SavedSearchesList;
