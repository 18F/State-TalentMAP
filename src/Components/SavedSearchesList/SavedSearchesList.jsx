import React from 'react';
import PropTypes from 'prop-types';
import SavedSearchesCard from '../SavedSearchesCard';
import { SAVED_SEARCH_PARENT_OBJECT } from '../../Constants/PropTypes';

const SavedSearchesList = ({ savedSearches, goToSavedSearch, deleteSearch }) => {
  const searchList = savedSearches.results.slice().map(savedSearch => (
    <SavedSearchesCard
      savedSearch={savedSearch}
      goToSavedSearch={goToSavedSearch}
      deleteSearch={deleteSearch}
      key={savedSearch.id}
    />
  ));
  return (
    <div className="usa-grid-full">
      {searchList}
    </div>
  );
};

SavedSearchesList.propTypes = {
  savedSearches: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
};

export default SavedSearchesList;
