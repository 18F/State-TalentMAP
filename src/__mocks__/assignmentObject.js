const assignmentObject = {
  start_date: '10-10-2016',
  estimated_end_date: '10-10-2018',
  position: 'Specialist',
};

export default assignmentObject;
