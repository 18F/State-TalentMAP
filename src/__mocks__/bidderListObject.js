const bidderListObject = {
  count: 3,
  results: [
    {
      id: 1,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Doe',
        email: 'doej@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 2,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'Jane',
        last_name: 'Doe',
        email: 'doejane@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 3,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Smith',
        email: 'smithj@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
  ],
};

export default bidderListObject;
