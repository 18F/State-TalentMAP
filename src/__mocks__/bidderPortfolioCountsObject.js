const bidderPortfolioCountsObject = {
  all: 100,
  bidding: 50,
  inpanel: 20,
  inpost: 30,
};

export default bidderPortfolioCountsObject;
