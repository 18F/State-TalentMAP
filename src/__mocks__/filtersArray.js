const filters = [
  { item: { description: 'language' }, data: [{ code: 'a', custom_description: 'letter A' }] },
  { item: { description: 'skill' }, data: [{ code: '1', custom_description: '1' }] },
  { item: { description: 'grade' }, data: [{ code: '2', custom_description: '2' }] },
  { item: { description: 'region' }, data: [{ code: 'bur', custom_description: 'bureau' }] }];

export default filters;
