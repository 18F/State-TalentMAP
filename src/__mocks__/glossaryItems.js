const glossaryItems = [
  { id: 2, title: 'Assignment Officer (AO)', definition: 'The employee responsible for filling the position.' },
  { id: 3, title: 'Bidder', definition: 'An employee who is applying for a position.' },
  { id: 4, title: 'Cable', definition: 'A secure text-based message sent from one State department location to another.' },
  { id: 5, title: 'Bureau', definition: 'The organization sponsoring a position. Most bureaus oversee regional areas, such as Western Europe. Others, such as the Bureau of Consular Affairs, oversee functional groups. If referring to a regional bureau, include the acronym -- State Department employees often use acronyms to refer to regional bureaus. In UI copy, put the acronym before the full name, eg: (WHA) Bureau of Western Hemispheric Affairs.' },
  { id: 6, title: 'Career development officer (CDO)', definition: 'An employee responsible for helping others find and successfully bid for appropriate positions for their career goals.' },
  { id: 7, title: 'Career Track', definition: 'A chosen professional competency.' },
  { id: 8, title: 'Cone', definition: 'A cone is a colloquial synonym for a generalist or specialist functional field. Bids can be "in cone" (for a position whose skill code belongs to the bidders\' specialization) or "out of cone." Each cone contains multiple skill codes.' },
  { id: 9, title: 'Cost of living allowance (COLA)', definition: 'A salary increase to reimburse employees for prices at a foreign post that are higher than same costs in Washington, DC. The specific value of the allowance changes based on local prices. COLA is a colloquial shorthand for what is officially known as post allowance.' },
];

export default glossaryItems;
