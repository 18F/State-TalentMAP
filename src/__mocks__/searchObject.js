const searchObject = {
  id: 1,
  name: 'search',
};

export default searchObject;
