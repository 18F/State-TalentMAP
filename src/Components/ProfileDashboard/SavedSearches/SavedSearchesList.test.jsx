import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SavedSearchesList from './SavedSearchesList';
import { searchObjectParent, mappedParams } from '../../../__mocks__/searchObject';

describe('SavedSearchesListComponent', () => {
  const props = {
    savedSearches: searchObjectParent,
    goToSavedSavedSearch: () => {},
    mappedParams,
    filtersIsLoading: true,
  };
  it('is defined', () => {
    const wrapper = shallow(<SavedSearchesList {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<SavedSearchesList {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when there are no bids', () => {
    const wrapper = shallow(<SavedSearchesList {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
