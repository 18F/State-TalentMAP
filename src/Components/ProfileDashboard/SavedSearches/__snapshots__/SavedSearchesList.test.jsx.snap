// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSearchesListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full saved-searches-list-profile-container"
>
  <Spinner
    filled={false}
    inverse={false}
    size="big"
    type="homepage-position-results"
  />
</div>
`;

exports[`SavedSearchesListComponent matches snapshot when there are no saved searches 1`] = `
<div
  className="usa-grid-full saved-searches-list-profile-container"
>
  <Spinner
    filled={false}
    inverse={false}
    size="big"
    type="homepage-position-results"
  />
</div>
`;
